x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/curations_logic.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { Meta } from '../../../../../common/types';
+import { DEFAULT_META } from '../../../shared/constants';
+
+import {
+  clearFlashMessages,
+  setSuccessMessage,
+  flashAPIErrors,
+} from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { EngineLogic } from '../engine';
+
+import { DELETE_MESSAGE, SUCCESS_MESSAGE } from './constants';
+import { Curation, CurationsAPIResponse } from './types';
+
+interface CurationsValues {
+  dataLoading: boolean;
+  curations: Curation[];
+  meta: Meta;
+}
+
+interface CurationsActions {
+  onCurationsLoad(response: CurationsAPIResponse): CurationsAPIResponse;
+  onPaginate(newPageIndex: number): { newPageIndex: number };
+  loadCurations(): void;
+  deleteCurationSet(id: string): string;
+}
+
+export const CurationsLogic = kea<MakeLogicType<CurationsValues, CurationsActions>>({
+  path: ['enterprise_search', 'app_search', 'curations_logic'],
+  actions: () => ({
+    onCurationsLoad: ({ results, meta }) => ({ results, meta }),
+    onPaginate: (newPageIndex) => ({ newPageIndex }),
+    loadCurations: true,
+    deleteCurationSet: (id) => id,
+  }),
+  reducers: () => ({
+    dataLoading: [
+      true,
+      {
+        loadCurations: () => true,
+        onCurationsLoad: () => false,
+      },
+    ],
+    curations: [
+      [],
+      {
+        onCurationsLoad: (_, { results }) => results,
+      },
+    ],
+    meta: [
+      DEFAULT_META,
+      {
+        onCurationsLoad: (_, { meta }) => meta,
+        onPaginate: (state, { newPageIndex }) => {
+          const newState = { page: { ...state.page } };
+          newState.page.current = newPageIndex;
+          return newState;
+        },
+      },
+    ],
+  }),
+  listeners: ({ actions, values }) => ({
+    loadCurations: async () => {
+      const { meta } = values;
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      try {
+        const response = await http.get(`/api/app_search/engines/${engineName}/curations`, {
+          query: {
+            'page[current]': meta.page.current,
+            'page[size]': meta.page.size,
+          },
+        });
+        actions.onCurationsLoad(response);
+      } catch (e) {
+        flashAPIErrors(e);
+      }
+    },
+    deleteCurationSet: async (id) => {
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+      clearFlashMessages();
+
+      if (window.confirm(DELETE_MESSAGE)) {
+        try {
+          await http.delete(`/api/app_search/engines/${engineName}/curations/${id}`);
+          actions.loadCurations();
+          setSuccessMessage(SUCCESS_MESSAGE);
```

Comment:
I want to note this UX is also different from the standalone UI, and I ran/rubber-duckied this by Davey and got his approval.

Previously, `deleteCurationSet` **manually** spliced/removed the deleted ID from the `curations` array. However, I saw this as problematic because it didn't take pagination into account: if you have a table of 10 items on page 1, and you delete the current table array, the table would go down to only 9 items without correctly loading in the next item from page 2.

IMO, the simplest and most obvious solution for this is to call `loadCurations()` again instead of manually attempting to anticipate the actual curations array. I consider the trade-off (xhr call/load time) well worth it (over possible de-synced items and pagination issues).

Reply:
Good call 👍